selenium-client/driver.py

```diff
@@ -0,0 +1,130 @@
+                self.driver = webdriver.Edge(executable_path=drivername)
+            else:
+                self.driver = webdriver.Edge()
+            
+        else:
+            raise Exception("A valid browser must be selected")
+        
+        
+    def run_experiment(self, uniqname="skorman", network="UMich WiFi", vpn=False, vpn_loc="Michigan", your_loc="Ann Arbor, MI, USA"):
+        self.driver.get("https://test.reethika.info/measure")
+        time.sleep(0.5)
+        # Fill uniqname field
+        uniqname_elem = self.driver.find_element(By.NAME, "email")
+        uniqname_elem.send_keys(uniqname)
+        time.sleep(0.5)
+        
+        # Select Desktop
+        devices = self.driver.find_elements(By.NAME, "device")
+        devices[1].click()
+        
+        # Fill network field
+        network_elem = self.driver.find_element(By.NAME, "network")
+        network_elem.send_keys(network)
+        
+        # Fill browser field
+        browser_elem = Select(self.driver.find_element(By.ID, "browser"))
+        browser_elem.select_by_value(self.browser)
+        
+        vpn_select_elem = self.driver.find_elements(By.NAME, "exp_type")
+        if not vpn:
+            vpn_select_elem[0].click()
+        else:
+            vpn_select_elem[1].click()
+            time.sleep(0.3)
+            self.driver.find_element(By.NAME, "location_vpn").send_keys(vpn_loc)
+            self.driver.find_element(By.NAME, "location_user").send_keys(your_loc)
+            
+        time.sleep(0.5)
+        
+        self.driver.find_element(By.XPATH, "/html/body/form/input[7]").click()
+        time.sleep(0.5)    
+
+        
+        self.wait_for_completion()
+        
+        
+    def wait_for_completion(self):
+        counter = 0
+        while self.driver.find_element(By.ID, "status").text != "Done" and counter < 60:
+            counter += 1
+            time.sleep(1)
+            
+        print("Browser:", self.browser, "finished!")
+        self.driver.quit()
+
+
+def run_browser_test(browser, options):
+    driver = StartDriver(browser)
+    driver.run_experiment(**options)
+
+
+def run_all_browser_tests(browsers: list, options):
+    threads: list[threading.Thread] = []
+    for browser in browsers:
+        browser_thread = threading.Thread(target=run_browser_test, args=[browser, options])
+        browser_thread.start()
+        threads.append(browser_thread)
+        
+    for thread in threads:
+        thread.join()
+        
+@click.command()
+@click.option('-u', '--uniqname', default='skorman', help='Uniqname of person running test')
+@click.option('-n', '--network', default='UMich WiFi', help='Network connected to (e.g. UMich WiFi, Comcast Home, T-Mobile, Verizon)')
+@click.option('--vpn', is_flag=True, help='Indicate whether running through a VPN')
+@click.option('-l', '--vpn_loc', default='Ann Arbor, MI, USA', help='Location of VPN (City, State, Country)')
```

Comment:
For this one, there shouldn't be a default value, and show throw error if `--vpn` flag is set to true but `-l` flag is not specified.